src/components/OpenFileDialog/OpenFileDialog.js

```diff
@@ -0,0 +1,444 @@
+            noFilteredDataText: i18n.t(
+                "No items found. Try adjusting your search or filter options to find what you're looking for."
+            ),
+            newButtonLabel: i18n.t('Create new'),
+        }
+
+        switch (type) {
+            case 'visualization': {
+                texts = {
+                    modalTitle: i18n.t('Open a visualization'),
+                    loadingText: i18n.t('Loading visualizations'),
+                    errorTitle: i18n.t("Couldn't load visualizations"),
+                    errorText: i18n.t(
+                        'There was a problem loading visualizations. Try again or contact your system administrator.'
+                    ),
+                    noDataText: i18n.t(
+                        'No visualizations found. Click New visualization to get started.'
+                    ),
+                    noFilteredDataText: i18n.t(
+                        "No visualizations found. Try adjusting your search or filter options to find what you're looking for."
+                    ),
+                    newButtonLabel: i18n.t('New visualization'),
+                }
+                break
+            }
+            case 'map': {
+                texts = {
+                    modalTitle: i18n.t('Open a map'),
+                    loadingText: i18n.t('Loading maps'),
+                    errorTitle: i18n.t("Couldn't load maps"),
+                    errorText: i18n.t(
+                        'There was a problem loading maps. Try again or contact your system administrator.'
+                    ),
+                    noDataText: i18n.t(
+                        'No maps found. Click New map to get started.'
+                    ),
+                    noFilteredDataText: i18n.t(
+                        "No maps found. Try adjusting your search or filter options to find what you're looking for."
+                    ),
+                    newButtonLabel: i18n.t('New map'),
+                }
+                break
+            }
+        }
+
+        return texts[key]
+    }
+
+    useEffect(() => {
+        // only fetch data when the dialog is open
+        if (open) {
+            refetch({
+                page,
+                sortField,
+                sortDirection,
+            })
+        }
+    }, [open, page, sortField, sortDirection])
+
+    useEffect(() => {
+        // reset pagination when filters are applied/changed
+        setPage(1)
+
+        refetch({
+            sortField,
+            sortDirection,
+            filters: formatFilters(filters),
+        })
+    }, [filters])
+
+    const headers = [
+        {
+            field: 'displayName',
+            label: i18n.t('Name'),
+        },
+        {
+            field: 'created',
+            label: i18n.t('Created'),
+        },
+        {
+            field: 'lastUpdated',
+            label: i18n.t('Last updated'),
+        },
+    ]
+
+    if (type === 'visualization') {
```

Comment:
I think you should set up consts for the supported types so we don't have to match on these strings everywhere.
e.g. `const TYPE_VISUALIZATION = 'visualization'`

Reply:
Done. See comment above.